### PR TITLE
Support relative paths/names in 'rename => newname()'

### PR DESCRIPTION
**Pull request by vpodzime**

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6007/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6007/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6007/

**Comment by vpodzime** [comment]
@cf-bottom jenkins again, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6008/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6008/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6008/

File: cf-agent/verify_files_utils.c [hunk 1]
```diff
@@ -2134,12 +2128,40 @@ static PromiseResult VerifyName(EvalContext *ctx, char *path, const struct stat
...
+    if (attr->rename.newname != NULL)
     {
+        char newname[CF_BUFSIZE];
+        if (IsAbsoluteFileName(attr->rename.newname))
+        {
+            strlcpy(newname, attr->rename.newname, CF_BUFSIZE);
+        }
+        else
+        {
+            strcpy(newname, path);
```
Comment (olehermanse) [inline cf-agent/verify_files_utils.c:2140]: I'd prefer `strncpy` with `sizeof(newname)` as the size argument.

**Review by olehermanse — APPROVED** [review]
LGTM 👍

**Review by larsewi — APPROVED** [review]

**Comment by larsewi** [comment]
Is there a memory leak here? Can't see `free(chrooted_path)` before `return`. Then again allocation to `chrooted_path` is conditional, and this may be a case where it's not allocated?

```c
if (attr->rename.disable_suffix)
{
    if (!JoinSuffix(newname, sizeof(newname), attr->rename.disable_suffix))
    {
        return result;
    }
}
else
{
    if (!JoinSuffix(newname, sizeof(newname), ".cfdisabled"))
    {
        return result;
    }
}
```

File: cf-agent/verify_files_utils.c [hunk 2]
```diff
@@ -2134,12 +2128,40 @@ static PromiseResult VerifyName(EvalContext *ctx, char *path, const struct stat
     }
 
     PromiseResult result = PROMISE_RESULT_NOOP;
-    if (attr->rename.newname)
+    if (attr->rename.newname != NULL)
     {
+        char newname[CF_BUFSIZE];
+        if (IsAbsoluteFileName(attr->rename.newname))
+        {
+            strlcpy(newname, attr->rename.newname, CF_BUFSIZE);
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply cf-agent/verify_files_utils.c:2136]: Good point, I've added the check.

**Comment by vpodzime** [comment]
> Is there a memory leak here?

Good catch! Fixed.

**Review by larsewi — APPROVED** [review]

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6009/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6009/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6009/

**Review by olehermanse — APPROVED** [review]

**Pull request merged**